src/misc/fs_utils_posix.cpp

```diff
@@ -162,11 +162,13 @@ constexpr int PermissionsRW = S_IWUSR | S_IWGRP | S_IWOTH | PermissionsRO;
 
 // Attributes 'hidden', 'system', and 'archive' are always taken from the
 // host extended attributes.
-constexpr uint8_t XattrReadMask = 0b0010'0110; // hidden, system, archive
-// Attributes 'read-only' and 'directory' are stored in extended attributes,
-// but not used by DOSBox, for these we are always using host file system
-// attribute bits.
-constexpr uint8_t XattrWriteMask = 0b0001'0001 | XattrReadMask;
+constexpr uint8_t XattrReadMask = 0b0010'0111; // read-only, hidden, system, archive
+// Attributes 'read-only' and 'directory' are stored in extended attributes.
+// For files, read-only is checked using host file system permissions.
+// For directories, we use the extended attribute.
```

Comment:
If I see correctly, for directories we are now going to use logical OR between attribute and file system flag, right?

Reply:
Done